langchain4j-voyage/src/main/java/dev/langchain4j/model/voyage/VoyageEmbeddingModel.java

```diff
@@ -0,0 +1,255 @@
+package dev.langchain4j.model.voyage;
+
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.model.embedding.DimensionAwareEmbeddingModel;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+import dev.langchain4j.model.output.Response;
+import dev.langchain4j.model.output.TokenUsage;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+import static dev.langchain4j.internal.RetryUtils.withRetry;
+import static dev.langchain4j.internal.Utils.getOrDefault;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+import static dev.langchain4j.model.voyage.VoyageApi.DEFAULT_BASE_URL;
+import static dev.langchain4j.model.voyage.VoyageEmbeddingModelName.VOYAGE_3_LITE;
+import static java.time.Duration.ofSeconds;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * An implementation of an {@link EmbeddingModel} that uses
+ * <a href="https://docs.voyageai.com/docs/embeddings">Voyage AI Embedding API</a>.
+ */
+public class VoyageEmbeddingModel extends DimensionAwareEmbeddingModel {
+
+    private final VoyageClient client;
+    private final Integer maxRetries;
+    private final VoyageEmbeddingModelName modelName;
+    private final String inputType;
+    private final Boolean truncation;
+    private final String encodingFormat;
+    private final int maxSegmentsPerBatch;
+
+    public VoyageEmbeddingModel(
+            String baseUrl,
+            Duration timeout,
+            Integer maxRetries,
+            String apiKey,
+            VoyageEmbeddingModelName modelName,
```

Comment:
```suggestion
            String modelName,
```
Let's not limit user

Reply:
Done.